gui/account/forms.py

```diff
@@ -320,6 +325,15 @@ def clean_bsdusr_home(self):
 
             if home.startswith('/mnt/'):
                 bsdusr_username = self.cleaned_data.get('bsdusr_username', '')
+                volumes = ['/mnt/{}'.format(volume.vol_name) for volume in Volume.objects.all()]
+
+                if len(zfs.list_datasets(path=user_home)) > 0 and \
+                        (len(zfs.list_datasets(path=home)) > 0 and home.startswith(user_home)):
```

Comment:
What if home = '/mnt/tank/foobar' and user_home = '/mnt/tank/foo'?

Wont the condition satisfy and change disallowed? While it should be allowed?

Perhaps it should be: home.startswith(user_home + '/'))

Or am I missing something?

Reply:
Also, you need to make sure the user is being modified and it is not a new user. e.g. wrap all this block with `if self.instance.id:`

Reply:
@william-gr You are right about the if statement, that didn't cross my mind.